api-server/src/server/middlewares/csurf.js

```diff
@@ -1,7 +1,7 @@
 import csurf from 'csurf';
 
 export const csrfOptions = {
-  domain: process.env.COOKIE_DOMAIN || 'localhost',
+  domain: process.env.COOKIE_DOMAIN,
```

Comment:
Help me think this through:
- We had an issue with this, because Safari was not correctly resolving a domain of `localhost`?
- Removing this, sets `domain` to `undefined`
- Setting `domain` to undefined, defaults it to the current domain

> https://www.npmjs.com/package/csurf#:~:text=domain%20%2D%20sets%20the%20domain%20the%20cookie%20is%20valid%20on(defaults%20to%20current%20domain).

Reply:
csurf just adds `domain` into the `set-cookie` header, so the behaviour is whatever `domain` does in that header. That's described on [mdn](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#domaindomain-value), specifically:

> If omitted, this attribute defaults to the host of the current document URL, not including subdomains.

If you check the cookie domains for `jwt_access_token` and `csrf_token` they should both be `localhost`, so I think it's defaulting correctly.